### PR TITLE
docs: Refactor README.md

### PR DESCRIPTION
**Pull request by lucydodo**
I've made a few improvements to the README.md I'd like to get some feedback on these improvements, so I'm leaving a PR.
I'd be grateful if anyone would feel free to review it. Thanks! 😄 

- Changes
	- Add a CodeQL badge
	- Grouping badges by category
	- Remove 'Table of Contents' section
		- This feature was added via the New UI on GitHub. so I'm removing it.
	- Add guidance for 'continuous' build
	- Add a way to install the nightly version for macOS with SQLCipher support
	- Rename Twitter to X
	- A few readability improvements

You can see the changes in the following: https://github.com/sqlitebrowser/sqlitebrowser/tree/docs-refactor-readme

### CODE REVIEW

**Comment by justinclift** [comment]
This looks decent, though I'm thinking we should keep the Table of Contents.

That seems like it's probably useful because we have lots of sections. :smile:

File: README.md [hunk 1]
```diff
@@ -90,31 +66,29 @@ Download Windows releases here:
...
 reported after the last release. :D
 
 ## macOS
 
-#### Stable release
-
 DB Browser for SQLite works well on macOS.
 
-* macOS 10.12 (Sierra) - 10.15 (Catalina) are tested and known to work.
+* macOS 10.15 (Catalina) - 14.0 (Sonoma) are tested and known to work.
```
Comment (justinclift) [inline README.md:89]: Updated & corrected macOS version numbers is good too. :smile:

**Comment by mgrojo** [comment]
> This looks decent, though I'm thinking we should keep the Table of Contents.
> 
> That seems like it's probably useful because we have lots of sections. 😄

I think @lucydodo is removing it because you can [display a TOC from the menu button](https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/). But I think that many people don't know that and will probably miss it. I knew that, but I always forget about it.

**Comment by justinclift** [comment]
> Ideally, GitHub should have a pragma to include an inline TOC ...

Oh.  That reminds me... Asciidoc (rather than markdown) has that capability.

For example, when the "WebAssembly" wiki page on the Go wiki became pretty long, I converted it from Markdown to Asciidoc specifically so it could autogenerate the ToC.

The first few lines of that file (in raw text mode) are now:

```
WebAssembly
===========
:toc:
:toc-title:
:toclevels: 2
:icons:

```

Which generates the Table of Contents seen here: https://github.com/golang/go/wiki/WebAssembly

That being said, I'm not really a fan of Asciidoc otherwise.  Markdown seems more natural (to me), though that's probably just a familiarity thing.

**Comment by lucydodo** [comment]
Correct. I removed the TOC because, as @mgrojo said, GitHub added that feature.
To be honest, I often forget about that feature too (I don't think it's very accessible).

I don't mind adding it back because it doesn't make it less readable and it's not that hard, what do you think?

**Comment by justinclift** [comment]
I reckon we should add it back.  That way it works for everyone that reads the page, and people don't have to know about special GitHub buttons. :smile:

**Comment by lucydodo** [comment]
Added a TOC. Any thoughts on the postion of TOC?

**Comment by justinclift** [comment]
Table of Contents placement looks good, and everything else seems fine too.

Happy to merge this, unless there's other stuff that needs doing to it? :smile:

**Comment by lucydodo** [comment]
Sure, Go ahead! :)

**Comment by justinclift** [comment]
Done. :smile:

**Pull request merged**